mitiq/pec/representations/learning.py

```diff
@@ -136,13 +223,55 @@ def depolarizing_noise_loss_function(
     return success, epsilon_opt
 
 
+def depolarizing_noise_loss_function(
+    epsilon: npt.NDArray[np.float64],
+    operations_to_mitigate: List[QPROGRAM],
+    training_circuits: List[QPROGRAM],
+    ideal_values: npt.NDArray[np.float64],
+    noisy_executor: Executor,
+    pec_kwargs: Dict["str", Any],
+    pec_data: npt.NDArray[np.float64] = np.array([]),
+    observable: Optional[Observable] = None,
+) -> float:
+    if pec_data.size > 0:
+        ind = np.abs(pec_data[:, 0] - epsilon).argmin()
+        mitigated_values = pec_data[ind, 1:]
+
+    else:
+        representations = [
+            represent_operation_with_local_depolarizing_noise(
+                operation,
+                epsilon[0],
+            )
+            for operation in operations_to_mitigate
+        ]
+        mitigated_values = np.array(
+            [
+                execute_with_pec(
+                    circuit=training_circuit,
+                    observable=observable,
+                    executor=noisy_executor,
+                    representations=representations,
+                    full_output=False,
+                    **pec_kwargs,
+                )
+                for training_circuit in training_circuits
+            ]
+        )
+
+    return np.mean(
+        abs(mitigated_values.reshape(-1, 1) - ideal_values.reshape(-1, 1)) ** 2
+    )
```

Comment:
I'm pretty sure this should do the same thing.
```suggestion
    return np.mean((mitigated_values - ideal_values) ** 2)
```

Reply:
That looks simpler, but it doesn't give the same result if `mitigated_values` is a column and `ideal_values` is a row or vice versa. We could enforce that they have to be the same shape, but I think this makes the function easier to use.

Reply:
Ah good catch, I didn't realize that they could be of different shapes. What situations would cause one to be a row and the other a column? I think we should try and be consistent there.

Reply:
Above suggestion works now that I've unpacked the output from each execution of `evaluate`, so that `ideal_values` has the same format as `mitigated_values`.